Signal/src/ViewControllers/SelectRecipientViewController.m

```diff
@@ -376,11 +376,17 @@ - (BOOL)hasValidPhoneNumber
     }
     NSString *possiblePhoneNumber =
         [self.callingCode stringByAppendingString:self.phoneNumberTextField.text.digitsOnly];
-    PhoneNumber *parsedPhoneNumber = [PhoneNumber tryParsePhoneNumberFromUserSpecifiedText:possiblePhoneNumber];
+    NSArray<PhoneNumber *> *parsePhoneNumbers =
+        [PhoneNumber tryParsePhoneNumbersFromsUserSpecifiedText:possiblePhoneNumber
+                                              clientPhoneNumber:[TSAccountManager localNumber]];
```

Comment:
This is one of those situations where an innocuous sounding helper method is going to create a db transaction (readonly).

I know it's not elegant, but memoizing `[StorageManager localNumber]` in `TSAccountManager` might be some good diligence.

take it or leave it.